### ISSUE TITLE
Data preprocessing

### ISSUE THREAD

**Issue by iLOSTA** [iLOSTA/The-Project#7]
as per the project plan, this week, week of June 25, I am supposed to work on data preprocessing. But I am ahead of the plan on this regard. Already finished with this phase and moved to the next which is ML Model Selection. Please refer to 'Multi-Label text classifiers'.

**Comment by iLOSTA** [comment]
Data preprocessing steps I have done:
- Reading the dataset from the csv file using Pandas.
- Performing text cleaning on 'X' including: low letters, remove stopwords and any special symbols.
- Using TfidfVectorizer (single word or bigram) on 'X' to convert words into number representations. 
- Using Multilabel Binarizer to convert labels into vectors of 0's and 1's.
- Manually split the dataset into training/testing sets (80/20).
- Tried 5/10 fold Cross Validation as another way of dividing the dataset into training/testing sets. (no noticable difference from the the 80/20 split).

**Comment by iLOSTA** [comment]
Tried concatenating the 5 responses from the API for each sample and the result is not good.
the accuracy of the NN classifier dropped from 99% to 94%.
 
-  Tried training the model for longer time (30 epochs compared to 15 previously) and the result slightly improved (96%) but the model was overfitting, which is not good.

- Tried with unigram, bigram and (1,2)gram, the results are the same; around 95%.

CONCLUSION: concatenating 5 in 1 does not work.
